packit_service/worker/tasks.py

```diff
@@ -74,3 +113,240 @@ def babysit_copr_build(self, build_id: int):
     """ check status of a copr build and update it in DB """
     if not check_copr_build(build_id=build_id):
         self.retry()
+
+
+# tasks for running the handlers
+@celery_app.task(name="task.run_copr_build_start_handler")
```

Comment:
Does it make sense to have enum for that? Or, what about using handler class names?

Just thinking if we can use fewer strings and make that more type-able...

Reply:
I am not sure what to do about this, I have not found any elegant way of getting rid of the strings :thinking:

Reply:
Task names need to be strings. We could leave them out and [change the automatic naming behaviour](https://docs.celeryproject.org/en/stable/userguide/tasks.html#changing-the-automatic-naming-behavior) so we have names that makes sense for us, though honestly I'm not sure the benefits this effort would bring us worth it.

Reply:
We can have string enums if we want.

My concerns about strings:

- hard to follow across the code (find usage,...)
- hard to rename (I know sed, but still.)
- fewer benefits from type-checkers

Reply:
`Task names need to be strings`

Yes, I meant it to get rid of hardcoding the strings, sorry. 

`I'm not sure the benefits this effort would bring us worth it.`
+1

Reply:
@lachmanfrantisek do you mean something like this?
```python3
class TaskName(str, enum.Enum):
    copr_build_start_handler = "task.run_copr_build_start_handler"
   ...
```

Reply:
@lbarcziova looks good!

> Task names need to be strings

I am not against strings as types but against the need of writing the same strings multiple times. Sorry if I didn't make that clear at the start.

Reply:
@lachmanfrantisek enum added, PTAL